### PR TITLE
Feat(server)/sign up api

### LINKED ISSUE #21: Server: API Sign Up

### PR DESCRIPTION
**Pull request by tunganh252**
Closes #21

### CODE REVIEW

**Comment by gialynguyen** [comment]
@tunganh252 Rebase to resolve the conflicts.

**Comment by tunganh252** [comment]
@gialynguyen I've resolved this conflicts. Can you review this "pull request" again, thanks.

**Comment by gialynguyen** [comment]
Merged #60. Rebase to resolve the conflicts.

File: apps/server/src/config/root.config.ts [hunk 1]
```diff
@@ -23,3 +23,6 @@ export const mongoDB: MongoDB = {
   database: process.env.YEMUSIC_DB_CONNECTION_STRING || 'mongodb://localhost/db_yemusic_test',
   mongoOptions: {},
 };
+
+/* Setting the expiration time for the refresh token (a date 10 days from now). */
+export const refreshTokenExpires: number = Date.now() + 24 * 60 * 60 * 10000;
```
Comment (trungpham71198) [inline apps/server/src/config/root.config.ts:28]: you should create a common variable for Auth.

File: apps/server/src/datasources/controllers/auth/mutation.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    /* It's updating the user and then finding the user again. */
+    let user = await User.findOne({ email: verifyInfo['email'] });
+    if (!user) throw { errorCode: 'E-04', message: 'An error occurred while verify email user.' };
+
+    const email = verifyInfo['email'];
+
+    /* It's creating a token and a refresh token. */
+    const token = yeToken.createToken({ payload: { email } });
+    const refreshToken = yeToken.createToken({
+      expiration: parseInt(process.env.JWT_REFRESH_EXPIRATION),
```
Comment (trungpham71198) [inline apps/server/src/datasources/controllers/auth/mutation.ts:96]: You should get JWT_REFRESH_EXPIRATION from config.

**Comment by tunganh252** [comment]
@trungpham71198 
I've resolved the problems that you commented on Sunday.
Can you review that again, please. Thanks

File: apps/server/src/datasources/controllers/auth/mutation.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { refreshTokenExpires, jwtConfig } from '../../../config';
+import { TSignUpInput } from '../../interface';
+import { User } from '../../models';
+import { authUtils } from '../../utils/controllers';
+import { yeToken } from '../../utils/controllers';
+import { nodemailer } from '../../utils/controllers/nodemailer';
+
+export async function signUp(req: Request, res: Response, next: NextFunction): Promise<Response> {
```
Comment (trungpham71198) [inline apps/server/src/datasources/controllers/auth/mutation.ts:10]: remove try-catch, use function useHttpHandler for all cases this.

**Review by khoaquocngo — APPROVED** [review]

**Pull request merged**